### PR TITLE
docs(dev): replace old cli docs with v2 cli docs

### PR DESCRIPTION
**Pull request by pcattori**
Closes: #

- [ ] Docs
- [ ] Tests

Testing Strategy:

### CODE REVIEW

File: docs/other-api/dev.md [hunk 1]
```diff
@@ -29,57 +22,366 @@ Builds your app for production. This command will set `process.env.NODE_ENV` to
...
+</docs-info>
+
+### With custom app server
+
+If you used a template to get started, hopefully it's already integrated with `remix dev` out-of-the-box.
+If not, you can follow these steps to integrate your project with `remix dev`:
+
+1. Replace your dev scripts in `package.json` and use `-c` to specify your app server command:
+
+```json
```
Comment (MichaelDeBoey) [inline docs/other-api/dev.md:71]:
```suggestion
```json filename=package.json
```

File: docs/other-api/dev.md [hunk 2]
```diff
@@ -29,57 +22,366 @@ Builds your app for production. This command will set `process.env.NODE_ENV` to
...
+{
+  "dev": "remix dev -c \"node ./server.js\""
```
Comment (MichaelDeBoey) [inline docs/other-api/dev.md:73]:
```suggestion
  "scripts": {
    "dev": "remix dev -c \"node ./server.js\""
  }
```

File: docs/other-api/dev.md [hunk 3]
```diff
@@ -29,57 +22,366 @@ Builds your app for production. This command will set `process.env.NODE_ENV` to
...
+
+The `remix dev --port` option sets the internal port used for hot updates.
+**It does not affect the port your app runs on.**
+
+To set your app server port, set it the way you normally would in production.
+For example, you may have it hardcoded in your `server.js` file.
+
+If you are using `remix-serve` as your app server, you can use its `--port` flag to set the app server port:
+
+```
```
Comment (MichaelDeBoey) [inline docs/other-api/dev.md:155]:
```suggestion
```sh
```

**Review by MichaelDeBoey — APPROVED** [review]

File: docs/pages/v2.md [hunk 4]
```diff
@@ -781,93 +781,10 @@ module.exports = {
 };
 ```
 
-## remix dev
```
Comment (brophdawg11) [inline docs/pages/v2.md:784]: I think we should keep this v2 migration guide in place as-is through the release of v2 (and likely for a long while after).  Folks are going to still be on 1.x for a while and will want access to a centralized migration guide to get to v2.

**Review by brophdawg11 — APPROVED** [review]

**Pull request merged**